starlite/plugins/tortoise_orm.py

```diff
@@ -0,0 +1,94 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Type, cast
+
+from tortoise.fields import ReverseRelation
+from tortoise.fields.relational import RelationalField
+
+from starlite import MissingDependencyException
+from starlite.plugins.base import PluginProtocol
+
+try:
+    from tortoise import Model, ModelMeta
+    from tortoise.contrib.pydantic import PydanticModel, pydantic_model_creator
+except ImportError as e:
+    raise MissingDependencyException("tortoise-orm is not installed") from e
+
+if TYPE_CHECKING:
+    from pydantic import BaseModel
+
+
+class TortoiseORMPlugin(PluginProtocol[Model]):
+    _models_map: Dict[Type[Model], Type[PydanticModel]] = {}
+    _data_models_map: Dict[Type[Model], Type[PydanticModel]] = {}
+
+    def _create_pydantic_model(self, model_class: Type[Model], **kwargs: Dict[str, Any]) -> Type[PydanticModel]:
+        """
+        Takes a tortoitse model_class instance and convert it to a subclass of the tortoise PydanticModel.
+        It fixes some issues with the result of the tortoise model creator.
+        """
+        pydantic_model = cast(Type[PydanticModel], pydantic_model_creator(model_class, **kwargs))
+        for (
+            field_name,
+            tortoise_model_field,
+        ) in model_class._meta.fields_map.items():  # pylint: disable=protected-access
+            if field_name in pydantic_model.__fields__:
+                if not tortoise_model_field.required:
+                    pydantic_model.__fields__[field_name].required = False
+                if tortoise_model_field.null:
+                    pydantic_model.__fields__[field_name].allow_none = True
+        return pydantic_model
+
+    def to_pydantic_model_class(self, model_class: Type[Model], **kwargs: Any) -> Type[PydanticModel]:
```

Comment:
seeing as you immediately pop `parameter_name` out of `kwargs` you could consider adding it as an explicit kwarg with a default `None`, e.g.,;

```python
def to_pydantic_model_class(self, model_class: Type[Model], parameter_name: Optional[str] = None, **kwargs)

Reply:
not possible im afraid - this causes it to not fit the protocol and all tools complain

Reply:
makes sense :+1: